src/ProblemDetailsNotFoundHandler.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-problem-details for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-problem-details/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\ProblemDetails;
+
+use Interop\Http\ServerMiddleware\DelegateInterface;
+use Interop\Http\ServerMiddleware\MiddlewareInterface as ServerMiddlewareInterface;
+use Negotiation\Negotiator;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Stratigility\Delegate\CallableDelegateDecorator;
+
+class ProblemDetailsNotFoundHandler implements ServerMiddlewareInterface
+{
+    /**
+     * @var ProblemDetailsResponseFactory
+     */
+    private $responseFactory;
+
+    /**
+     * @param null|ProblemDetailsResponseFactory $responseFactory Factory to create a response to
+     *     update and return when returning an 404 response.
+     */
+    public function __construct(ProblemDetailsResponseFactory $responseFactory = null)
+    {
+        $this->responseFactory = $responseFactory ?: new ProblemDetailsResponseFactory();
+    }
+
+    /**
+     * Creates and returns a 404 response.
+     *
+     * @param ServerRequestInterface $request Ignored.
+     * @param DelegateInterface $delegate Ignored.
```

Comment:
Hm... missed before, sorry. These params description is wrong. These params are not ignored at all. We can remove that part of docs block to not duplicate with method signature.

Reply:
I can do this on merge.

One other thing I've discovered recently: return types in PHP are _covariant_, which means you can specify a return type _more specific_ than the one defined in the interface or parent class. That means we can add `: ResponseInterface` here safely to make it self-documenting. :smile: